index.js

```diff
@@ -37,10 +34,30 @@ var anymatch = function(criteria, value, returnIndex, startIndex, endIndex) {
     default:
       result = false;
     }
-    if (result) { matchIndex = index + startIndex; }
+    if (result) {
+      matchIndex = index + startIndex;
+    }
     return result;
   }
-  var matched = criteria.slice(startIndex, endIndex).some(testCriteria);
+  var crit = criteria;
+  var negGlobs = crit.reduce(function(arr, criterion, index) {
+    if (typeof criterion === 'string' && criterion[0] === '!') {
+      if (crit === criteria) {
+        // make a copy before modifying
+        crit = crit.slice();
+      }
+      crit[index] = null;
+      arr.push(criterion.substr(1));
+    }
+    return arr;
+  }, []);
+  if (!negGlobs.length || !micromatch.any(string, negGlobs)) {
+    if (path.sep === '\\' && typeof string === 'string') {
```

Comment:
The `isWindows` variable storing the first result may bring some perf improvement, since anymatch aims to be fast.